lib/oneview-sdk/image-streamer/resource/api300/build_plan.rb

```diff
@@ -17,6 +17,38 @@ module API300
       # Build Plan resource implementation for Image Streamer
       class BuildPlan < Resource
         BASE_URI = '/rest/build-plans'.freeze
+
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::ImageStreamer::Client] client The client object for the Image Streamer appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          super
+          # Default values:
+          @data['type'] ||= 'OeBuildPlan'
+          @data['customAttributes'] ||= []
+        end
+
+        # Build step of the build plan.
+        # @param build_step [Array]  The Build step array of the build plan
+        def set_build_step(build_step = [])
+          set('buildStep', build_step)
+          @data['buildStep'].each do |step|
+            step = Hash[step.map { |k, v| [k.to_s, v] }]
+            raise IncompleteResource, 'Please set the planScriptUri attribute!' unless step['planScriptUri']
+            set_custom_attributes(step['planScriptUri'])
+          end
+        end
+
+        private
+
+        # Sets the custom attributes.
+        # @param [String]  plan_script_uri uri of the plan script
+        def set_custom_attributes(plan_script_uri)
+          plan_script = OneviewSDK::ImageStreamer::API300::PlanScripts.find_by(@client, uri: plan_script_uri)
```

Comment:
@jsmartt , do not be surprised. This test would really fail because I made the change today, but I could not test why HW in India was not available when hw come back, I would run the test to make sure it was working.